### PR TITLE
Add interface for native camera

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Add interface to create a traceLocation from the native camera

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6071

### CODE REVIEW

**Review by whiskey — APPROVED** [review]
Thanks for the quick response. This should help with my task!

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift [hunk 1]
```diff
@@ -108,5 +111,41 @@ struct TraceLocation {
 		let encodedByteRepresentation = byteRepresentation.base32EncodedString
 		return String(format: "https://e.coronawarn.app/c1/%@", encodedByteRepresentation).uppercased()
 	}
+}
+
+extension TraceLocation {
+	init?(qrCodeString: String) {
+		
+		guard let data = qrCodeString.base32DecodedData else {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift:119]: please move it inside the struct TraceLocation and use the usual marks

**Review by KaiTeuber — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift]: We need the extension as we want to keep the normal init valid and to initialize the struct with the normal properties, i added the MARK

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift [hunk 2]
```diff
@@ -31,5 +31,49 @@ struct TraceLocation {
...
+			Log.error("Couldn't serialize the data")
+			return nil
+		}
+		Log.debug("Data found: \(String(describing: data))")
+
 
+		do {
+			// creates a fake event for the moment
+			let qRCodePayload = try SAP_Internal_Pt_QRCodePayload(serializedData: data)
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift:51]:
```suggestion
			let qrCodePayload = try SAP_Internal_Pt_QRCodePayload(serializedData: data)
```

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift [hunk 3]
```diff
@@ -31,5 +31,49 @@ struct TraceLocation {
...
+			}
+			self = TraceLocation(
+				id: id,
+				version: Int(traceLocation.version),
+				type: TraceLocationType(traceLocationTypeProtobuf: eventInformation.type),
+				description: traceLocation.description_p,
+				address: traceLocation.address,
+				startDate: startDate,
+				endDate: endDate,
+				defaultCheckInLengthInMinutes: Int(eventInformation.defaultCheckInLengthInMinutes),
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift:70]: 0 should be converted to `nil` her as well

File: src/xcode/ENA/ENA/Source/RootCoordinator.swift [hunk 4]
```diff
@@ -125,14 +125,18 @@ class RootCoordinator: RequiresAppDependencies {
...
 		tabBarController.selectedIndex = index
-
+		
+		guard let route = Route(guid),
+			  case let Route.checkin(key) = route,
+			  let traceLocation = TraceLocation(qrCodeString: key)
+		else {
+			Log.error("URL didn't match the route", log: .checkin, error: nil)
+			return
+		}
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/RootCoordinator.swift:135]: Should better be checked before we switch to the event tab I guess

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**